python/lsst/daf/persistence/posixStorage.py

```diff
@@ -780,6 +780,57 @@ def writeFitsCatalogStorage(butlerLocation, obj):
         return
 
 
+def readMatplotlibStorage(butlerLocation):
+    """Read from a butlerLocation (always fails for this storage type).
+
+    Parameters
+    ----------
+    butlerLocation : ButlerLocation
+        The location & formatting for the object(s) to be read.
+
+    Returns
+    -------
+    A list of objects as described by the butler location. One item for
+    each location in butlerLocation.getLocations()
+    """
+    raise NotImplementedError("Figures saved with MatplotlibStorage cannot be retreived using the Butler.")
+
+
+def writeMatplotlibStorage(butlerLocation, obj):
+    """Writes a matplotlib.figure.Figure to a location, using the template's
+    filename suffix to infer the file format.
+
+    Parameters
+    ----------
+    butlerLocation : ButlerLocation
+        The location & formatting for the object to be written.
+    obj : matplotlib.figure.Figure
+        The object to be written.
+
+    Raises
+    ------
+    ValueError
+        Raised if the file
+    """
+    additionalData = butlerLocation.getAdditionalData()
+    locations = butlerLocation.getLocations()
+    with SafeFilename(os.path.join(butlerLocation.getStorage().root, locations[0])) as locationString:
+        logLoc = LogicalLocation(locationString, additionalData)
+        # SafeFilename appends a random suffix, which corrupts the extension
+        # matplotlib uses to guess the file format.
+        # Instead, we strip off the extension from the original location
+        # and pass that as the format directly.
+        _, ext = os.path.splitext(locations[0])
+        if ext:
+            ext = ext[1:]  # strip off leading '.'
+        else:
+            # If there is no extension, we let matplotlib fall back to its
+            # default.
+            ext = None
+        obj.savefig(logLoc.locString(), format=ext)
+        return
```

Comment:
Is the **return** meant to be indented to within the **with SafeFilename** here?

Reply:
The empty `return` here serves no purpose, and should be removed.